android/src/main/java/com/rivereactnative/RiveReactNativeView.kt

```diff
@@ -23,9 +25,46 @@ import com.facebook.react.uimanager.ThemedReactContext
 import com.facebook.react.uimanager.events.RCTEventEmitter
 import java.io.UnsupportedEncodingException
 
+class ReactNativeRiveViewLifecycleObserver() :
+  DefaultLifecycleObserver {
+  override fun onCreate(owner: LifecycleOwner) { }
+
+  override fun onStart(owner: LifecycleOwner) { }
+
+  override fun onResume(owner: LifecycleOwner) { }
+
+  override fun onPause(owner: LifecycleOwner) { }
+
+  override fun onStop(owner: LifecycleOwner) { }
+
+  /**
+   * OnDestroy is different in Rive ReactNative compared to Rive Android.
+   * Releasing dependencies is managed in `ReactNativeRiveAnimationView.dispose()`
+   *
+   * TODO: Expose a way in Rive Android to access the list of dependencies to dispose so that
+   * these can always be in sync. Or override `RiveViewLifecycleObserver`.
+   */
+  @CallSuper
+  override fun onDestroy(owner: LifecycleOwner) {
+    owner.lifecycle.removeObserver(this)
+  }
+}
+
+class ReactNativeRiveAnimationView(private val context: ThemedReactContext) : RiveAnimationView(context) {
+  private lateinit var dependencies: List<RefCount>;
+
+  fun dispose() {
+    dependencies.forEach { it.release() }
+  }
+
+  override fun createObserver(): LifecycleObserver {
+    dependencies = listOfNotNull(controller, rendererAttributes.assetLoader)
```

Comment:
Ah ok - so the todo is basically to hook into one source of truth for these dependencies, rather than copying what RiveAnimationView does here, is that right?

Reply:
Exactly! Once RN uses the latest Android.